Code/Tools/FBuild/FBuildCore/Graph/ObjectNode.cpp

```diff
@@ -205,6 +222,114 @@ ObjectNode::~ObjectNode()
     return true;
 }
 
+// DeterminePreBuildDynamicDependenciesNeedToBuild
+//------------------------------------------------------------------------------
+/*virtual*/ bool ObjectNode::DeterminePreBuildDynamicDependenciesNeedToBuild(bool forceClean) const
+{
+	if( GetPreBuildPreprocessor() == nullptr )
+		return false;
+
+	if ( forceClean ||
+		 GetSourceFile()->GetStamp() > GetStamp() ||
+		 GetPreBuildPreprocessor()->GetStamp() > GetStamp())
+	{
+		return true;
+	}
+	return false;
+}
+
+// DoPreBuildDynamicDependencies
+//------------------------------------------------------------------------------
+/*virtual*/ Node::BuildResult ObjectNode::DoPreBuildDynamicDependencies(Job * job)
+{
+	m_PreBuildDynamicDependencyNames.Clear();
+
+	Args fullArgs;
+
+	{
+		Array< AString > tokens(1024, true);
```

Comment:
This could be combined into BuildArgs. Originally I broke it out so that there would be less impact on established code, but looking at it now it may make more sense to use BuildArgs and take advantage of other compiler specific substitutions if needed. At least it would be consistent. Thoughts?